drivers/dma/dma_nxp_dummy_dma.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/dma.h>
+#include <zephyr/logging/log.h>
+
+/* used for driver binding */
+#define DT_DRV_COMPAT nxp_dummy_dma
+
+/* macros used to parse DTS properties */
+#define IDENTITY_VARGS(V, ...) IDENTITY(V)
+
+#define _DUMMY_DMA_CHANNEL_INDEX_ARRAY(inst)\
+	LISTIFY(DT_INST_PROP_OR(inst, dma_channels, 0), IDENTITY_VARGS, (,))
+
+#define _DUMMY_DMA_CHANNEL_DECLARE(idx) {}
+
+#define DUMMY_DMA_CHANNELS_DECLARE(inst)\
+	FOR_EACH(_DUMMY_DMA_CHANNEL_DECLARE,\
+		 (,), _DUMMY_DMA_CHANNEL_INDEX_ARRAY(inst))
+
+LOG_MODULE_REGISTER(nxp_dummy_dma);
+
+/* a few notes about how this driver:
+ *
+ * 1) This is used by SOF, but shouldn't impose
+ * any restriction regarding the applications using it.
+ * Specifically, this is used by SOF on host side. The
+ * driver needs to expose a DMA-like interface because
```

Comment:
In the context of SOF actually 'host' side is the side that runs Linux. 

![image](https://github.com/zephyrproject-rtos/zephyr/assets/62316/d371027b-d770-4b36-b8d9-0ccc43ba7f5f)

Maybe you can say that this is used by SOF to copy data from host side.